Sources/DockProgress/DockProgress.swift

```diff
@@ -32,6 +32,7 @@ public final class DockProgress {
 		case bar
 		/// TODO: Make `color` optional when https://github.com/apple/swift-evolution/blob/master/proposals/0155-normalize-enum-case-representation.md is shipping in Swift
 		case circle(radius: Double, color: NSColor)
+		case badge(color: NSColor, badgeLabel: String)
```

Comment:
Having the `badeLabel` as a `String` means it's impossible to update it during its run, which is the whole point of having it in the first place. We could either make it a closure that is expected to return a string or add a new optional property to the DockProgress instance that is only used for the `badge` style. Thoughts on which one to pick?

Reply:
You're right. I'd go with closure since that seems like a better approach to me. I'll look into it.

Reply:
Alright. Closure it is.

Reply:
Hmmm now that I've implemented it with I closure I'm thinking it might not be the most user-friendly thing.. Like you'd have to call it like this (for example) right?
```swift
.badge(color: .systemBlue, badgeLabel: {
				return "\(Int(DockProgress.progressValue * 100))"
			})
```